cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,21 +1,41 @@
+import _ = require("cypress/types/lodash");
+import {MenuContentPage} from "../page/index";
+import {ProductsListPage} from "../page/index";
+import {ShopingCartPage} from "../page/index";
+import {LoginPage} from "../page/index";
+import {AddressStepPage} from "../page/index";
+import {ShippingStepPage} from "../page/index";
+import {PaymentStepPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductsListPage();
+const shopingCartPage = new ShopingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
 describe("Buy a t-shirt", () => {
-  it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get(
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-
-    cy.get("#SubmitLogin").click();
-    cy.get(".cart_navigation button").click();
-    cy.get("#cgv").check();
-    cy.get("#form button").click();
-    cy.get("#HOOK_PAYMENT a.bankwire").click();
+  it("then should be bought a t-shirt", () => {
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.clickAddCartBtn();
+    productsListPage.clickProceedCheckoutBtn();
+
+    shopingCartPage.clickProceedCheckoutBtn();
+
+    loginPage.typeEmailUser();
+    loginPage.typePasswordUser();
+    loginPage.clickProceedSubmitLoginBtn();
```

Comment:
you can have just one single method, called login or signIn, which receives as parameter the email and password of the user to log in, so we can reuse that page object to login any user